### DISCUSSION TITLE
Increasing fidelity of tesselation from raster tesselation

### DISCUSSION THREAD

**Discussion by darrencpagan** [neperfepx/neper#556] [Q&A]
Hi,

I'm looking for guidance to improve the fidelity of tessellations. Using the below tesr file:
![cycle5_test_tesr](https://user-images.githubusercontent.com/7535678/189565254-72b5b122-4630-4c1b-b153-d6b32c884162.png)

and the following tessellation call:

```
neper -T -n from_morpho -domain 'cube(1,1,1)' -morpho 'tesr:file(cycle5.tesr)' \
            -morphooptiobjective 'tesr:pts(region=all,res=15)+val(bounddist)' \
            -morphooptistop itermax=100 -reg 1 -o cycle5_test
```
,
I get the following tessellation:
![cycle5_test_tess](https://user-images.githubusercontent.com/7535678/189565503-56b9c580-bc3e-4667-9dca-e5c81ab3f460.png)
I was wondering if there are better options I could be using to increase fidelity between the tesr and tess file.

Thanks,

**Comment by rquey** [comment]
You are not using the best options.

Generally speaking, first thing to care about is the domain. Make sure it is properly defined.

Then, I would probably leave `-morphoobjective` and `-morphooptistop` to their default values. Your `-morphooptiobjective` does increase the resolution, but your `-morphooptistop` stops optimization *very* early. Neper will typically use thousands of iterations, and this specific (difficult) case, which has very elongated grains, will need even more.  I would recommend to let Neper converge as much as possible. Seeing the default output would be interesting.

Is it a single-scale or a multi-scale microstructure, including grain subdivisions?

**Comment by darrencpagan** [comment]
Thank you, do you have a suggestion about the domain? What might the problem with the definition be? Also, I will run with more iterations.

Currently this is a single scale microstructure, but there may be ways to make it multiple scale with prior beta information.

**Reply by rquey** [reply]
About the domain, just make sure that it is correct (not having your tesr at hand, I cannot say if this is the case).

**Comment by darrencpagan** [comment]
Thank you for the help. Got much better results with some domain cleaning (very small cell removal) and many more iterations (10k-100k).
![cycle5_test_tesr](https://user-images.githubusercontent.com/7535678/190465873-08eb6b22-6653-4b1e-8af1-19d6ee34a485.png)
![cycle5_test_tess](https://user-images.githubusercontent.com/7535678/190465899-ae5056c7-6088-45f7-972d-72378f44580a.png)

Do you think further improvements could be gained from making the domains two scale? The large scale structure would be prior beta grains and finer scale structure would alpha colonies in a Ti alloy?

**Reply by rquey** [reply]
Yes, small cell removal definitely helps.

In principle, yes, if the polycrystal is 2-scale, it has to go through 2-scale tessellation, although I have never done it (but it is an interesting case). I am afraid that the tessellation won't be very similar to the polycrystal, however, since there is significant non-convexity... It will work only if you want or are happy with a simplified morphology (which is what Neper does)...
